include/seqan3/alphabet/detail/convert.hpp

```diff
@@ -39,7 +39,7 @@ constexpr std::array<out_t, alphabet_size<in_t>> convert_through_char_representa
     [] () constexpr
     {
         std::array<out_t, alphabet_size<in_t>> ret{};
-        for (typename in_t::rank_type i = 0; i < alphabet_size<in_t>; ++i)
+        for (detail::min_viable_uint_t<alphabet_size<in_t>> i = 0; i < alphabet_size<in_t>; ++i)
```

Comment:
```suggestion
        for (decltype(alphabet_size<in_t>) i = 0; i < alphabet_size<in_t>; ++i)
```

Reply:
I thought these are eqivalent

Reply:
If I make that change, the bam test compiles forever :scream:

Reply:
`for (auto i = decltype(alphabet_size<in_t>){0}; i < alphabet_size<in_t>; ++i)` funktioniert wieder ^^

Reply:
> If I make that change, the bam test compiles forever scream

It should actually be faster. Keep in mind that `detail::min_viable_uint_t<` instantiates a type, whereas decltype does not.

> for (auto i = decltype(alphabet_size<in_t>){0}; i < alphabet_size<in_t>; ++i) 

whut?

Reply:
:woman_shrugging: I dunno :woman_shrugging:

Reply:
This is still weird and it is not guaranteed that the min_viable_uint_type is the correct type. Just because we decide to do that doesn't mean other alphabets don't use `size_t` so this will result in conversion. I don't understand why decltype should cause problems here. If it does, please document it in the code.

Reply:
hm if decltype is more correct, i can use the second version?
```
for (auto i = decltype(alphabet_size<in_t>){0}; i < alphabet_size<in_t>; ++i)
```

Reply:
yes

Reply:
alright. I changed it and made a comment that it does not work the other way